src/kOS/Suffixed/Vector.cs

```diff
@@ -11,34 +11,46 @@ public class Vector : Structure
         public double Y { get; set; }
         public double Z { get; set; }
 
-        public Vector(Vector3d init)
-        {
-            X = init.x;
-            Y = init.y;
-            Z = init.z;
-        }
-
-        public Vector(Vector3 init) // Vector3 is the single-precision version of Vector3d.
-        {
-            X = init.x;
-            Y = init.y;
-            Z = init.z;
-        }
+        public Vector(Vector3d init) :this(init.x,init.y,init.z) { }
+        public Vector(Vector3 init) :this(init.x, init.y, init.z) { }
+        public Vector(float x, float y, float z) : this((double)x,(double)y,(double)z) { }
 
         public Vector(double x, double y, double z)
         {
             X = x;
             Y = y;
             Z = z;
+            InitializeSuffixes();
         }
 
-        public Vector(float x, float y, float z)
+        private void InitializeSuffixes()
         {
-            X = x;
-            Y = y;
-            Z = z;
+            AddSuffix("X", new SetSuffix<double>(() => X, value => X = value));
+            AddSuffix("Y", new SetSuffix<double>(() => Y, value => Y = value));
+            AddSuffix("Z", new SetSuffix<double>(() => Z, value => Z = value));
```

Comment:
by the time you make it to setting something, I have already clobbered the type with the following line 

https://github.com/KSP-KOS/KOS/blob/develop/src/kOS.Safe/Encapsulation/Suffixes/SetSuffix.cs#L17

if it can convert it will be converted there